src/toil/lib/toilDocker.py

```diff
@@ -0,0 +1,213 @@
+"""
+    Module for calling Docker. Assumes `docker` is on the PATH.
+    Contains two user-facing functions: dockerCall and dockerCheckOutput
+    Uses Toil's defer functionality to ensure containers are shutdown even in case of job or pipeline failure
+    Example of using dockerCall in a Toil pipeline to index a FASTA file with SAMtools:
+        def toil_job(job):
+            work_dir = job.fileStore.getLocalTempDir()
+            path = job.fileStore.readGlobalFile(ref_id, os.path.join(work_dir, 'ref.fasta')
+            parameters = ['faidx', path]
+            dockerCall(job, tool='quay.io/ucgc_cgl/samtools:latest', work_dir=work_dir, parameters=parameters)
+"""
+# from builtins import str
+# from builtins import map
+import logging
+import os
+import pipes
+import subprocess
+import docker
+import base64
+import time
+from tarfile import TarFile
+
+from docker.errors import ContainerError
+from docker.errors import ImageNotFound
+from docker.errors import APIError
+from docker.errors import NotFound
+from docker.utils.types import LogConfig
+from docker.api.container import ContainerApiMixin
+
+from bd2k.util.retry import retry
+from docker import client
+from pwd import getpwuid
+
+from toil.lib import dockerPredicate
+from toil.lib import FORGO
+from toil.lib import STOP
+from toil.lib import RM
+from toil.test import timeLimit
+
+_logger = logging.getLogger(__name__)
+
+
+def dockerCheckOutput(job, tool, parameters=None, workDir=None, dockerParameters=None, deferParam=None,
+                      checkOutput=True, containerName=None):
```

Comment:
There ought to be a docstring here. It also might make sense to just dump `**kwargs` into dockerCall, so all the options (like `data_dir`) are supported.

Reply:
Good point.  I'll make the change.